<a href="https://colab.research.google.com/github/Twujira/TigistData110/blob/main/Markov_Chain.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## Markov_Chain Project1

By: Tigist Wujira

Introduction

As a child, I used to bet with my siblings every night on who would host the evening news. The prime time news began streaming at 8:00 pm on Ethiopia's national TV, Ethiopian Television (ETV). There were three well-known hosts who took turns leading the news, but it was impossible to predict who would be on air that evening.

Their name is Hermela, Amare, and Temesgen

Transaction Matrix:

     Hermela | Amare | Temesgen
    H|  0.7  | 0.2   |  0.1
    A|  0.3  | 0.5   |  0.2
    T|  0.2  | 0.2   |  0.6



In [1]:
#Import necessary packages
from sympy import Matrix
import numpy as np

In [2]:
# Define the transition matrix
T_host = np.matrix([
    [0.7, 0.2, 0.1],
    [0.3, 0.5, 0.2],
    [0.2, 0.2, 0.6]
])

# Print the transition matrix
print(T_host)


[[0.7 0.2 0.1]
 [0.3 0.5 0.2]
 [0.2 0.2 0.6]]


As an initial state, if the host were Hermela. v= [1,0,0]

In [3]:
# Define the initial state vector
V0 = np.array([1, 0, 0])

# Multiply the initial state vector with the transition matrix
V1 =V0*T_host

# Print the result
print(V1)


[[0.7 0.2 0.1]]


For Day2

In [4]:
#multiply day1 result with transition matrix
V2=V1*T_host
print(V2)

[[0.57 0.26 0.17]]


For Day3

In [5]:
#multiply day2 result with transition matrix
V3=V2*T_host
print(V3)

[[0.511 0.278 0.211]]


## Activity 2

Given 2022 Population

x0=

    [600,000]

    [400,000]

In [14]:
#Define transax matrix
A = np.array([
    [0.92,0.04],
    [0.08,0.96]
    ])
print(A)



[[0.92 0.04]
 [0.08 0.96]]


In [15]:
#Initial 2022 population
x0 = np.matrix([
    [600000],
    [400000]
])
print(x0)

[[600000]
 [400000]]


## 1. For 2023

In [16]:
#Population predict 2023

P_2023=A*x0
print(P_2023)

[[568000.]
 [432000.]]


## 2. For 2030

The difference between 2030 and 2022 is 8 years

Therefore:

'A' the power of 8

P_2030= A^8*x0 (the initial population)

In [17]:
A_8 = np.linalg.matrix_power(A, 8)
print(A_8)

[[0.57308968 0.21345516]
 [0.42691032 0.78654484]]


In [18]:
P_2030=A_8*x0
print(P_2030)

[[429235.87328147]
 [570764.12671853]]


Conclusion

By the year 2030, the city is expected to have a population of 429,236, while the suburbs are projected to have a population of 570,764.